frontend/src/views/CellGuide/components/common/OntologyDagView/components/AnimatedNodes/index.tsx

```diff
@@ -82,12 +82,14 @@ export default function AnimatedNodes({
     event: React.MouseEvent<HTMLDivElement>,
     datum: TreeNodeWithState
   ) => {
-    const id = setTimeout(() => {
-      track(EVENTS.CG_TREE_NODE_HOVER, {
-        cell_type: datum.name,
-      });
-    }, 2 * 1000);
-    setTimerId(id);
+    if (!timerId) {
```

Comment:
do we need to setTimerId(null) after timeout callback function is called?

e.g.,

```ts
setTimeout(() => {
        track(EVENTS.CG_TREE_NODE_HOVER, {
          cell_type: datum.name,
        });
        setTimerId(null); // <--- this?
      }, 2 * 1000);
```

Reply:
Also can we write a test for this please 🙏 ?

Reply:
No we don't, because the mouseout function handles that.

Reply:
Can you point me to an example where we have an e2e test for analytics? That would be helpful for me.

Reply:
I don't think we have an example yet, so you'll be the first one!

My initial thought is that I think we can detect console.log, since we do log that when `track()` is called

Reply:
> No we don't, because the mouseout function handles that.

Ah I see!  I was thinking about colocating the create and destroy in the same place, so we don't run into this kind of confusion. What do you think?

Reply:
We still need to destroy in handleMouseOut anyway to handle the case where users exit out of the node before the setTimeout triggers. Which means that adding setTimerId(null) in the setTimeout is redundant.

Reply:
Ah good point, because we want to cancel it within 2 seconds. Got it thank you!